streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -234,10 +256,10 @@ private ClientResponse sendRequest(final ClientRequest clientRequest) {
     /**
      * Fetch the metadata for all topics
      */
-    public Collection<MetadataResponse.TopicMetadata> fetchTopicsMetadata() {
+    public MetadataResponse fetchMetadata() {
 
         final ClientRequest clientRequest = kafkaClient.newClientRequest(
-            getBrokerId(),
+            getAnyReadyBrokerId(),
```

Comment:
In fact the current logic in `prepareTopic` loops until `each one of the topic metadata has been propagated to at least one broker`. So the above is correct.